### PR TITLE
Improve version check

### PR DESCRIPTION
**Pull request by balloob**
- Set a max version that we won't support (2.0)
- Fail if server is max version or higher

### CODE REVIEW

File: zwave_js_server/const.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 from typing import Dict, List
 
 MIN_SERVER_VERSION = "1.0.0-beta.2"
```
Comment (MartinHjelmare) [inline zwave_js_server/const.py:5]: Is this min version still ok? The add-on runs beta 4 now.

File: zwave_js_server/client.py [hunk 2]
```diff
@@ -86,16 +87,19 @@ async def connect(self) -> None:
...
-        if cur_version < min_version:
+        cur_version = AwesomeVersion(self.version.server_version)
+        bad_version = None
+
+        if cur_version.strategy != AwesomeVersionStrategy.SEMVER:
+            bad_version = f"Failed to parse Z-Wave JS Server version {self.version.server_version}"
+        elif cur_version < AwesomeVersion(MIN_SERVER_VERSION):
+            bad_version = f"Z-Wave JS Server needs to be at least version {MIN_SERVER_VERSION}. Found {self.version.server_version}"
+        elif cur_version >= AwesomeVersion(MAX_SERVER_VERSION):
+            bad_version = f"Z-Wave JS Server is newer or equal to {MAX_SERVER_VERSION}. Found {self.version.server_version}"
```
Comment (MartinHjelmare) [inline zwave_js_server/client.py:98]:
```suggestion
            bad_version = (
                "Failed to parse Z-Wave JS Server version "
                f"{self.version.server_version}"
            )
        elif cur_version < AwesomeVersion(MIN_SERVER_VERSION):
            bad_version = (
                f"Z-Wave JS Server needs to be at least version {MIN_SERVER_VERSION}. "
                f"Found {self.version.server_version}"
            )
        elif cur_version >= AwesomeVersion(MAX_SERVER_VERSION):
            bad_version = (
                f"Z-Wave JS Server is newer or equal to {MAX_SERVER_VERSION}. "
                f"Found {self.version.server_version}"
            )
```

File: zwave_js_server/client.py [hunk 3]
```diff
@@ -7,9 +7,10 @@
 from typing import Any, Dict, Optional, cast
 
 from aiohttp import ClientSession, ClientWebSocketResponse, WSMsgType, client_exceptions
-from packaging.version import parse as parse_version
+from awesomeversion import AwesomeVersion, AwesomeVersionException
```
Comment (MartinHjelmare) [inline zwave_js_server/client.py:10]: The exception is never raised in awesomeversion.
```suggestion
from awesomeversion import AwesomeVersion
```

File: zwave_js_server/const.py (hunk 1 above)
Comment (balloob) [reply zwave_js_server/const.py:5]: I don't know. We are compatible with beta 2+ right ? It's not about the bugs, it's about the protocol.

File: zwave_js_server/client.py (hunk 3 above)
Comment (balloob) [reply zwave_js_server/client.py:10]: Yeah, it's coming in the next version that is being worked on .

File: zwave_js_server/const.py (hunk 1 above)
Comment (marcelveldt) [reply zwave_js_server/const.py:5]: beta4 solved the missing keys in the state dumps so maybe require that as minimum

File: zwave_js_server/client.py (hunk 3 above)
Comment (balloob) [reply zwave_js_server/client.py]: Next version is hitting a delay. Just going to remove this :)

**Review by MartinHjelmare — APPROVED** [review]

File: zwave_js_server/const.py (hunk 1 above)
Comment (MartinHjelmare) [reply zwave_js_server/const.py:5]: Let's bump this separately.

**Pull request merged**